### DISCUSSION TITLE
Add relying party / authorization server support?

### DISCUSSION THREAD

**Discussion by arianvp** [panva/oauth4webapi#111] [Ideas]
This library looks great.  I was wondering if there were plans to add building blocks for creating a relying party / authorization server side of things not just the client side of things?   e.g.:

1. helpers to build authorization handler / token handler
2. not only generate PKCE challenges but also functions to verify them
3. Not only generate DPoP signatures, but also functions to verify them

It'd be great to have building blocks to quickly set up a secure FAPI2.0 compatible authorization server in e.g. Deno or a Cloudflare worker.

**Comment by arianvp** [comment]
I know you also maintain https://github.com/panva/node-oidc-provider but that's focused on node and not WebAPI, and also is not typescript.

**Reply by panva** [reply]
Hi @arianvp 

> authorization server

no, absolutely not a chance.

> relying party

that's generally what an oauth 2.0 client is, did you mean resource server?

> resource server

https://github.com/panva/oauth4webapi/blob/v2.10.3/docs/functions/validateJwtAccessToken.md
